quinn/src/endpoint.rs

```diff
@@ -346,27 +346,31 @@ impl EndpointInner {
                 Poll::Ready(Ok(msgs)) => {
                     self.recv_limiter.record_work(msgs);
                     for (meta, buf) in metas.iter().zip(iovs.iter()).take(msgs) {
-                        let data = buf[0..meta.len].into();
-                        match self
-                            .inner
-                            .handle(now, meta.addr, meta.dst_ip, meta.ecn, data)
-                        {
-                            Some((handle, DatagramEvent::NewConnection(conn))) => {
-                                let conn =
-                                    self.connections
-                                        .insert(handle, conn, self.udp_state.clone());
-                                self.incoming.push_back(conn);
+                        for buf in buf[0..meta.len].chunks(meta.gso_size.unwrap_or(meta.len)) {
+                            let data: BytesMut = buf.into();
```

Comment:
I wonder if it would be more efficient to convert the entire slice to a `BytesMut` up front, then split it with `split_off` or similar, rather than making separate allocations.

Reply:
That was actually what I did at first before switching to `chunks()`, then convinced myself I didn't like the while loop. I've now gone back to doing only one allocation but perf wise it doesn't seem to make a difference (these are all same-sized relatively small buffers so I'm assuming that the allocator is being smart)

Reply:
If there's empirically no difference then I'm happy either way.